release_dashboard/test/src/test_state_generator.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:conductor_core/conductor_core.dart';
+import 'package:conductor_core/proto.dart' as pb;
+
+const String kConductorVersion = 'v1.0';
+const String kReleaseChannel = 'beta';
+const String kReleaseVersion = '1.2.0-3.4.pre';
+const String kEngineCandidateBranch = 'flutter-1.2-candidate.3';
+const String kFrameworkCandidateBranch = 'flutter-1.2-candidate.4';
+const String kWorkingBranch = 'cherrypicks-$kEngineCandidateBranch';
+const String kDartRevision = 'fe9708ab688dcda9923f584ba370a66fcbc3811f';
+const String kEngineCherrypick1 = 'a5a25cd702b062c24b2c67b8d30b5cb33e0ef6f0';
+const String kEngineCherrypick2 = '94d06a2e1d01a3b0c693b94d70c5e1df9d78d249';
+const String kEngineCherrypick3 = '94d06a2e1d01a3b0c693b94d70c5e1df9d78d255';
+const String kFrameworkCherrypick = '768cd702b691584b2c67b8d30b5cb33e0ef6f0';
+const String kEngineStartingGitHead = '083049e6cae311910c6a6619a6681b7eba4035b4';
+const String kEngineCurrentGitHead = '23otn2o3itn2o3int2oi3tno23itno2i3tn';
+const String kEngineCheckoutPath = '/Users/engine';
+const String kFrameworkStartingGitHead = 'df6981e98rh49er8h149er8h19er8h1';
+const String kFrameworkCurrentGitHead = '239tnint023t09j2039tj0239tn';
+const String kFrameworkCheckoutPath = '/Users/framework';
+final String kEngineLUCIDashboard = luciConsoleLink(kReleaseChannel, 'engine');
+final String kFrameworkLUCIDashboard = luciConsoleLink(kReleaseChannel, 'flutter');
+
+/// Helper function that generates a test conductor state.
+///
+/// Default state has all the info complete and valid with no cherrypick conflicts.
+pb.ConductorState generateConductorState({
+  bool? engineCherrypicksInConflict,
+  bool? frameworkCherrypicksInConflict,
+  String? conductorVersion = kConductorVersion,
```

Comment:
Are these constants used anywhere else? If not, let's remove them as it's preferred for defaults to be directly changed in the function signature.

Reply:
Yes, they are used in some of the tests. Could we keep the defaults? `generateConductorState` is used in a lot of the tests, and most of them use the default values. Otherwise I would have to pass the defaults every time.

Reply:
Ok sgtm